rs/messaging/tests/memory_tests.rs

```diff
@@ -146,15 +162,102 @@ fn check_guaranteed_response_message_memory_limits_are_respected_impl(
     // One extra tick to make sure everything is gc'ed.
     fixture.tick();
 
-    // Check the system agrees on 'no hanging calls'.
-    if fixture.open_call_contexts_count().values().sum::<usize>() != 0 {
-        return fixture.failed_with_reason("found call contexts after shutdown phase");
+    // Check the records agree on 'no hanging calls'.
+    if fixture
+        .canisters()
+        .into_iter()
+        .map(|canister| extract_metrics(&fixture.force_query_records(canister)))
+        .any(|metrics| metrics.hanging_calls != 0)
+    {
+        return fixture.failed_with_reason("found hanging calls in the records");
     }
 
     // After the fact, all memory is freed and back to 0.
     fixture.expect_guaranteed_response_message_memory_taken_at_most("Final check", 0, 0)
 }
 
+/// Runs a state machine test with two subnets, a local subnet with 2 canisters installed and a
+/// remote subnet with 5 canisters installed. All canisters, except one local canister referred to
+/// as `migrating_canister` is stopped.
+///
+/// In the first phase a number of rounds are executed on both subnets, including XNet traffic with
+/// the `migrating_canister` making random calls to all installed canisters (since all calls are
+/// rejected except those to self, downstream calls are disabled).
+///
+/// For the second phase, `migrating_canister` stops making calls and is then migrated to the
+/// remote subnet. Since all other canisters are stopped, there are bound to be a number of reject
+/// signals for requests in the reverse stream to the local_subnet. But since we migrated the
```

Comment:
Nit: We use "reverse stream" when we're dealing with pairs of streams (and one is obviously the "[outbound] stream" and the other the "reverse stream"). In this case, you should probably use "stream to the local subnet", "incoming stream(s)" or "inbound stream(s)".